python-threatexchange/threatexchange/exchanges/impl/stop_ncii_api.py

```diff
@@ -84,29 +82,26 @@ class StopNCIISignalExchangeAPI(
 
     def __init__(
         self,
-        subscription_key: t.Optional[str],
-        fetch_function_key: t.Optional[str],
-        additional_function_keys: t.Optional[t.Dict[api.StopNCIIEndpoint, str]] = None,
+        collab: CollaborationConfigBase,
+        client: api.StopNCIIAPI,
     ) -> None:
         super().__init__()
-        self._api = None
-        if subscription_key is not None and fetch_function_key is not None:
-            self._api = api.StopNCIIAPI(
-                subscription_key,
-                fetch_function_key,
-                additional_function_keys,
-            )
+        self.collab = collab
+        self.api = client
 
-    @property
-    def api(self) -> api.StopNCIIAPI:
-        if self._api is None:
-            raise Exception("StopNCII.org access tokens not configured.")
-        return self._api
+    @classmethod
```

Comment:
blocking: We're using classmethod here, but the type in SignalExchangeAPI is staticmethod.

What's mypy good for? 🤷🏼

Reply:
I went back and forth on whether it should be static or class (mostly trying to avoid the devilish complexity of typing the class correctly). 

Will catch you in followup conversation, but currently tending towards classmethod if we do follow through with auth.

mpypy doesn't care because the thing that comes out the other end is typing equivalent - same inputs and same outputs.